test/lit/validation/eh-nameless.wast

```diff
@@ -0,0 +1,29 @@
+;; Test for a nameless block containing a pop, which is allowed.
+
+;; XXXXXXXXXXXXXXX RUN: not wasm-opt --enable-reference-types %s -o - -S 2>&1 | filecheck %s --check-prefix NO-GC
+;; RUN:     wasm-opt --enable-reference-types --enable-gc %s -o - -S | filecheck %s --check-prefix GC
+
+;; NO-GC: all used types should be allowed
+
+;; GC:   (func $foo (type $0) (param $x eqref)
+
+(module
+ (tag $tag (param i32))
+
+ (func $0
+  (try
+   (do
+    (nop)
+   )
+   (catch $tag
+    (drop
+     (block (result i32)
+      (pop i32)
+     )
+    )
+   )
```

Comment:
This Wasm doesn't really make sense from a typing perspective unless the block takes an i32 parameter, which we don't yet support. I'm not sure we should _want_ to be able to parse something like this.

Reply:
Similarly, we probably should not be printing text that looks like this.

I'm working on using IRBuilder in the binary parser now, after which we will be able to update it to support control flow parameters, so we could fix this example to have proper typing, but even then, IRBuilder will lower the control flow parameters away using locals, so we still wouldn't be able to parse this IR directly.

Reply:
Yeah, I see what you mean. But this is consistent with our other extensions to the text format, e.g., this validates:
```wat
(module
  (func $foo
    (local $x (ref func))
    (block
      (local.set $x
        (ref.func $foo)
      )
    )
    (drop
      (local.get $x)
    )
  )
)
```
That isn't valid wat since the block should end the range of the non-nullable local, but we decided to allow it.

With that said, I do see that this change would be more intrusive and confusion, potentially. So maybe this PR is a bad idea.

But we do have a general problem here, where more passes then we expected need EH fixups. Modifying them all adds clutter and overhead, so I was hoping for a more general solution...

Reply:
> This Wasm doesn't really make sense from a typing perspective unless the block takes an i32 parameter, which we don't yet support. I'm not sure we should _want_ to be able to parse something like this.

I don't think I understand. Why does this not make sense unless the block takes an i32 input parameter? What does this have to do with an input parameter?

Reply:
`pop i32` models popping a value off the stack and then pushing it right back onto the stack to be consumed by its parent expression, so it's similar to a `nop`, except that it requires a value to be on the stack in the first place. In this example, there would only be a value available on the stack if the `block` took an `i32` parameter.

In the long run, I hope that the text and binary parsers will be able to lower legacy EH directly to new EH IR, so all the problems with pop validation will become moot. Maybe there are less intrusive changes we can make in the short term?